crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -147,14 +147,18 @@ def services
         end
 
         # Initiate the services shutdown by calling scripts on all nodes.
-        # For each cluster, it is enough to initiate the shutdown from one node (e.g. founder)
-        NodeObject.find("state:crowbar_upgrade AND pacemaker_founder:true").each do |node|
-          node.ssh_cmd("/usr/sbin/crowbar-shutdown-services-before-upgrade.sh")
-        end
-        # Shutdown the services for non clustered nodes
-        NodeObject.find("state:crowbar_upgrade AND NOT run_list_map:pacemaker-cluster-member").
-          each do |node|
-          node.ssh_cmd("/usr/sbin/crowbar-shutdown-services-before-upgrade.sh")
+        NodeObject.find("state:crowbar_upgrade").each do |node|
+          if node.roles.include?("pacemaker-cluster-member")
```

Comment:
Can we write the body of the `#each` block like this?

    node.shutdown_services

Then we can move the code for shutting down services on a single node into the `shutdown_services` instance method. Then later, HA nodes could implement service shutdown differently to non-HA nodes, either by subclasses (ugly), or by concerns (nicer).